rust/connlib/tunnel/src/client.rs

```diff
@@ -974,12 +1023,15 @@ impl ClientState {
             }
             ResourceDescription::Internet(_) => {}
         }
-
-        self.resources_by_id.insert(new_resource.id(), new_resource);
     }
 
     #[tracing::instrument(level = "debug", skip_all, fields(?id))]
     pub(crate) fn remove_resource(&mut self, id: ResourceId) {
+        self.disable_resource(id);
+        self.resources_by_id.remove(&id);
+    }
+
+    fn disable_resource(&mut self, id: ResourceId) {
```

Comment:
Can we rename `cidr_resources` to `active_cidr_resources` with this change? (And update its docs)